torchgeo/datasets/dota.py

```diff
@@ -0,0 +1,505 @@
+# Copyright (c) Microsoft Corporation. All rights reserved.
+# Licensed under the MIT License.
+
+"""DOTA dataset."""
+
+import os
+from collections.abc import Callable
+from typing import Any, ClassVar
+
+import matplotlib.pyplot as plt
+import numpy as np
+import pandas as pd
+import torch
+from matplotlib import patches
+from matplotlib.figure import Figure
+from PIL import Image
+from torch import Tensor
+
+from .errors import DatasetNotFoundError
+from .geo import NonGeoDataset
+from .utils import (
+    Path,
+    check_integrity,
+    download_url,
+    extract_archive,
+    percentile_normalization,
+)
+
+
+class DOTA(NonGeoDataset):
+    """DOTA dataset.
+
+    The `DOTA <https://captain-whu.github.io/DOTA/index.html>`__ is a large-scale object
+    detection dataset for aerial imagery containing RGB and gray-scale imagery from Google Earth, GF-2 and JL-1 satellites
+    as well as additional aerial imagery from CycloMedia. There are three versions of the dataset: V1.0, V1.5, and V2.0, where,
+    V1.0 and V1.5 have the same images but different annotations, and V2.0 extends both the images and annotations with more samples
+
+    Dataset features:
+
+    * 1869 samples in V1.0 and V.1.5 and 2423 samples in V2.0
+    * multi-class object detection (15 classes in V1.0 and V1.5 and 18 classes in V2.0)
+    * horizontal and oriented bounding boxes
+
+    Dataset format:
+
+    * images are three channel PNGs with various pixel sizes
+    * annotations are text files with one line per bounding box
+
+    Classes:
```

Comment:
Should these be ints? They are in a slightly different order than on the website